src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -323,6 +296,29 @@ create_reference_table(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * EnsureTableCanBeCreated checks if
+ * - we are on the coordinator
+ * - Citus version is correct
+ * - the current user is the owner of the table
+ * - relation kind is supported
+ */
+static void
+EnsureTableCanBeCreated(Oid relationOid)
+{
+	EnsureCoordinator();
```

Comment:
`CheckCitusVersion ` should be the first thing that we do on any UDF execution. So, few notes: It should be first line in this function if we want to keep it here. 

Second, I'd rather remove `CheckCitusVersion ` and have them explicitly on the UDFs. Because, if someone changes where `EnsureTableCanBeCreated ` called, `CheckCitusVersion ` could not do what it intends. 

In addition to that, I don't think `CheckCitusVersion ` fits into `EnsureTableCanBeCreated ` very well. `CheckCitusVersion` should prevent any UDF call (or anything on Citus) if the versions don't match. So, it is probably better to keep this explicitly per UDF, not directly relating to  `EnsureTableCanBeCreated `

Reply:
That makes sense, thanks for the suggestion and explanation